print.py

```diff
@@ -0,0 +1,102 @@
+import sys
+import os
+import time
+from threading import Lock
+
+from ftxwebsocket.client import FtxWebsocketClient
+from ftxrest.client import FtxClient
+
+
+if len(sys.argv) < 2:
+    raise ValueError("Must specify a market")
+
+MARKET = sys.argv[1]
+
+last_printed_trade = None
+ws_trades = []
+print_lock = Lock()
+rest_client = FtxClient(
+    api_key="Doesnt matter, just using public APIs",
+    api_secret="Doesnt matter, just using public APIs",
+)
+
+
+def _fetch_missing_trades_when_appropriate():
+    last_trade = _get_last_ws_trade()
+
+    with print_lock:
+        while not _get_last_ws_trade() or _get_last_ws_trade() == last_trade:
```

Comment:
This method is used to RESTfully fetch data that may have been missed when a websocket connection was lost. It is triggered via an `on_disconnect` callback based to the websocket client. Rather than making requests as soon as the websocket connection is lost, we wait until the websocket connection is restored before retrieving missed records. This way, we can be confident that there are no gaps between content fetched RESTfully and content fetched via the websocket.